test/integration/data-types.test.js

```diff
@@ -274,3 +274,30 @@ describe(Support.getTestDialectTeaser('DataTypes'), function() {
     testFailure(Type);
   });
 });
+
+describe(Support.getTestDialectTeaser('DataTypes') + " MySQL", function() {
```

Comment:
I am running this test for all dialects which supports `GEOMETRY`. Will wait for CI to finish to see if something goes wrong on other supported databases.